### PR TITLE
Add 4 cases for options test

### PR DESCRIPTION
**Pull request by sitoliu**
ID: 1585582 1624809 1624808 1631348
Signed-off-by: Sitong Liu <siliu@redhat.com>

### CODE REVIEW

**Comment by sitoliu** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/1751

**Comment by sitoliu** [comment]
(1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.indirect_desc.opt_on: PASS (255.93 s)
 (2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.indirect_desc.opt_off: PASS (255.68 s)
 (3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.mrg_rxbuf.opt_on: PASS (256.82 s)
 (4) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.mrg_rxbuf.opt_off: PASS (256.14 s)
 (5) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_30: PASS (50.06 s)
 (6) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_50: PASS (50.31 s)
 (7) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_on.mq_opt_on: PASS (51.49 s)
 (8) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_on.mq_opt_off: PASS (50.04 s)
 (9) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_on.mq_opt_on: PASS (56.88 s)
 (10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_on.mq_opt_off: PASS (57.59 s)

File: qemu/tests/cfg/nic_opt.cfg [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            variants:
+                - opt_on:
+                    mrg_rxbuf = on
+                - opt_off:
+                    mrg_rxbuf = off
+        - poll-us:
+            check_type = ping
+            variants:
+                - opt_30:
+                    vhost = on
```
Comment (xiywang) [inline qemu/tests/cfg/nic_opt.cfg]: Move this line below line 45? Because opt_30 and opt_50 share the same config

**Comment by vivianQizhu** [comment]
@xiywang @pezhang Please help review, thanks.

File: qemu/tests/cfg/nic_opt.cfg [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    netperf_output_unit = m
+    netperf_local_cpu = yes
+    netperf_remote_cpu = yes
+    netperf_para_sessions = 1
+    test_protocols = TCP_STREAM
+    netperf_client = ${main_vm}
+    netperf_server = localhost
+    netperf_package_sizes = 1024
+    sub_type = netperf_stress
+    sub_exit_timeout = 10
```
Comment (pezhang) [inline qemu/tests/cfg/nic_opt.cfg]: @sitoliu I would suggest put vhost=on here. As vhost=on is default option of virtio-net from libvirt layer.

File: qemu/tests/cfg/nic_opt.cfg [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                - opt_on:
+                    nic_extra_params = ",mrg_rxbuf=on"
+                - opt_off:
+                    nic_extra_params = ",mrg_rxbuf=off"
+        - poll-us:
+            check_type = ping
+            vhost = on
+            variants:
+                - opt_30:
+                    netdev_extra_params = ",poll_us=30"
```
Comment (pezhang) [inline qemu/tests/cfg/nic_opt.cfg]: @sitoliu Option is wrong, should be poll-us=30

File: qemu/tests/cfg/nic_opt.cfg [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                - opt_50:
+                    netdev_extra_params = ",poll_us=50"
```
Comment (pezhang) [inline qemu/tests/cfg/nic_opt.cfg]: same as above.

File: qemu/tests/nic_opt.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            ifname = utils_net.get_windows_nic_attribute(session=session,
+                                                         key="netenabled", value=True,
+                                                         target="netconnectionID")
+        else:
+            ifname = utils_net.get_linux_ifname(session,
+                                                vm.get_mac_address())
+        dest = utils_net.get_host_ip_address(params)
+
+        for size in package_sizes:
+            error_context.context("Test ping from '%s' to host '%s' on guest '%s'"
```
Comment (pezhang) [inline qemu/tests/nic_opt.py]: a little little suggestion, should add a space at the end of the line?

**Comment by sitoliu** [comment]
@pezhang Thanks, update it.

File: qemu/tests/cfg/nic_opt.cfg [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            variants:
+                - opt_on:
+                    nic_extra_params += ",mrg_rxbuf=on"
+                - opt_off:
+                    nic_extra_params += ",mrg_rxbuf=off"
+        - poll-us:
+            check_type = ping
+            variants:
+                - opt_30:
+                    netdev_extra_params_nic1 += ",poll-us=30"
```
Comment (pezhang) [inline qemu/tests/cfg/nic_opt.cfg]: @sitoliu Seems this option doesn't work as expected. As the qemu command line doesn't add this option like below:

# cat 05-Host...qcow2.virtio_scsi.up.virtio_net.Guest....x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_30/debug.log

    -device virtio-net-pci,mac=9a:66:67:68:69:6a,id=idzsfQY4,vectors=4,netdev=idMET9s7,bus=pci.0,addr=0x4  \
    -netdev tap,id=idMET9s7,vhost=on,vhostfd=22,fd=9 \

**Comment by sitoliu** [comment]
(01/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.indirect_desc.opt_on: PASS (320.12 s)
 (02/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.indirect_desc.opt_off: PASS (316.82 s)
 (03/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mrg_rxbuf.opt_on: PASS (318.76 s)
 (04/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mrg_rxbuf.opt_off: PASS (319.19 s)
 (05/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_30: PASS (219.87 s)
 (06/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_50: PASS (220.11 s)
 (07/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_on.mq_opt_on: PASS (223.70 s)
 (08/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_on.mq_opt_off: PASS (222.12 s)
 (09/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_off.mq_opt_on: PASS (219.37 s)
 (10/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_off.mq_opt_off: PASS (220.45 s)
 (11/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_on.mq_opt_on: PASS (261.60 s)
 (12/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_on.mq_opt_off: PASS (275.45 s)
 (13/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_off.mq_opt_on: PASS (258.62 s)
 (14/14) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_off.mq_opt_off: PASS (256.79 s)

File: qemu/tests/nic_opt.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                                  " with package size %d. " %
+                                  (ifname, dest, vm.name, size))
```
[the remark this answers is not in the document]
Comment (sitoliu) [reply qemu/tests/nic_opt.py]: Thanks, update it.

**Comment by pezhang** [comment]
Test pass.
4 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.indirect_desc.opt_on.q35 		PASS 	325.17
5 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.indirect_desc.opt_off.q35 		PASS 	282.53
6 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mrg_rxbuf.opt_on.q35 		PASS 	287.83
7 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mrg_rxbuf.opt_off.q35 		PASS 	282.67
8 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_30.q35 		PASS 	177.49
9 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_50.q35 		PASS 	235.61
10 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_on.mq_opt_on.q35 		PASS 	195.55
11 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_on.mq_opt_off.q35 		PASS 	184.54
12 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_off.mq_opt_on.q35 		ERROR 	82.85 	Failed to determine interface name with ... 	debug.log
13 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_off.mq_opt_off.q35 		ERROR 	138.61 	Failed to determine interface name with ... 
14 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_on.mq_opt_on.q35 		PASS 	240.35
15 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_on.mq_opt_off.q35 		PASS 	192.00
16 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_off.mq_opt_on.q35 		ERROR 	78.58 	Failed to determine interface name with ... 	debug.log
17 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_off.mq_opt_off.q35 		ERROR 	133.60 	Failed to determine interface name with ... 	debug.log
18 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.indirect_desc.opt_on.q35 		PASS 	379.70
19 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.indirect_desc.opt_off.q35 		PASS 	332.35
20 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mrg_rxbuf.opt_on.q35 		PASS 	318.77
21 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mrg_rxbuf.opt_off.q35 		PASS 	323.86
22 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_30.q35 		PASS 	232.57
23 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.poll-us.opt_50.q35 		PASS 	224.20
24 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_on.mq_opt_on.q35 		PASS 	231.60
25 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_on.mq_opt_off.q35 		PASS 	231.67
26 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_off.mq_opt_on.q35 		PASS 	231.32
27 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_2.ctrl_vq_off.mq_opt_off.q35 		PASS 	238.65
28 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_on.mq_opt_on.q35 		PASS 	239.33
29 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_on.mq_opt_off.q35 		PASS 	249.20
30 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_off.mq_opt_on.q35 		PASS 	234.25
31 	Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.nic_opt.mqcontrol.smp_64.ctrl_vq_off.mq_opt_off.q35 		PASS 	233.90 

Note: For ctrl_vq_off with rhel guest, there is a product id: 1631683. 

Looks good to me.

Acked by: Pei Zhang pezhang@redhat.com

File: qemu/tests/cfg/nic_opt.cfg [hunk 8]
```diff
@@ -0,0 +1,70 @@
+- nic_opt:
+    type = nic_opt
+    only virtio_net
+    only x86_64 ppc64 ppc64le
+    # ping test setting
```
[the remark this answers is not in the document]
Comment (sitoliu) [reply qemu/tests/cfg/nic_opt.cfg]: Could you help update this info to polarion cases? Thanks!

File: qemu/tests/cfg/nic_opt.cfg [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                    mq = off
+            variants:
+                - ctrl_vq_on:
+                    ctrl_vq = on
+                - ctrl_vq_off:
+                    ctrl_vq = off
+            variants:
+                - smp_2:
+                    smp = 2
+                - smp_64:
```
[the remark this answers is not in the document]
Comment (sitoliu) [reply qemu/tests/cfg/nic_opt.cfg]: 1/ `-smp = 1` takes no effect with kar. 
2/ What do you mean 

>  when the cpu is not enough in host?

In this case, no need to pin all the vcpus to physical cpus. Thanks.

**Pull request closed without merging**